omod-1.10/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_10/DrugController1_10Test.java

```diff
@@ -95,14 +96,20 @@ public void getDrugsByMapping_shouldReturnADrugThatMatchesTheCodeAndSourceAndThe
 		SimpleObject results = deserialize(handle(newGetRequest(getURI(), new Parameter("s", "getDrugsByMapping"),
 		    new Parameter("code", "CD41003"), new Parameter("source", sourceUuid), new Parameter("preferredMapTypes",
 		            mapTypeUuids))));
-		
+
 		assertEquals(0, Util.getResultsSize(results));
 		
 		mapTypeUuids = conceptService.getConceptMapType(2).getUuid();
 		results = deserialize(handle(newGetRequest(getURI(), new Parameter("s", "getDrugsByMapping"), new Parameter("code",
 		        "CD41003"), new Parameter("source", sourceUuid), new Parameter("preferredMapTypes", mapTypeUuids))));
-		
-		assertEquals(expectedDrugUuid, PropertyUtils.getProperty(Util.getResultsList(results).get(1), "uuid"));
+
+		// Checking if expected drug is contained in response of the request above
+		HashSet<Object> setOfDrugs=new HashSet<Object>();
+		for(Object drug:Util.getResultsList(results)) {
+			setOfDrugs.add(PropertyUtils.getProperty(drug,"uuid"));
+		}
+		assert setOfDrugs.contains(expectedDrugUuid);
```

Comment:
@Andu033 it seems your idea is correct, but can you confirm that whether pre or post-1.10.3 it is always the same set of drugs that is returned? Just ordered differently.

Unfortunately REST WS does not auto-format (yet), so you'll have to stick to the rules. See the third point in **8.2 Blank Spaces** [here](https://www.oracle.com/technetwork/java/javase/documentation/codeconventions-141388.html). With some renaming I would rather do:
```java
Set<Object> actualDrugs = new HashSet<Object>();
for(Object drug : Util.getResultsList(results)) {
  actualDrugs.add(PropertyUtils.getProperty(drug, "uuid"));
}
assert actualDrugs.contains(expectedDrugUuid);
```

Reply:
@mks-d  yes is the same set of drugs , i tested. The person who did the test wanted to test if 2 he get the drug from request but he saw that he got 2 drugs because 2 drugs matched these properties and said ok here is my drug the second one but he didn t checked for all versions .  And even if there was a different set of drugs each time my approach is still better because i check if the drug is returned by the request so if the drug exist in both places , he only check if the drug is the second one which is not what he wanted from the name of test (ShouldReturnADrugThatMaches....) a drug so he check if this return a drug that matches . He dont want to see if the drug is second he only want to see if the mapping return a drug . So my approach match 100% the test case. (if the drug is not returned by mapping fail, if it is pass and i also checked for other versions 1.10.3 post and pre).